beard/clustering/wrappers.py

```diff
@@ -0,0 +1,137 @@
+
+    linkage_ : ndarray
+        The linkage matrix.
+    """
+
+    def __init__(self, method="single", affinity="euclidean", threshold=None,
+                 n_clusters=2, criterion="distance", depth=2, R=None,
+                 monocrit=None):
+        """Initialize.
+
+        Parameters
+        ----------
+        :param method: string
+            The linkage algorithm to use.
+            See scipy.cluster.hierarchy.linkage for further details.
+
+        :param affinity: string
+            The distance metric to use. If affinity='precomputed', then
+            assume that X is a distance matrix.
+            See scipy.cluster.hierarchy.linkage for further details.
+
+        :param threshold: float or None
+            The thresold to apply when forming flat clusters.
+            See scipy.cluster.hierarchy.fcluster for further details.
+
+        :param n_clusters: int
+            The number of flat clusters to form, if threshold=None.
+
+        :param criterion: string
+            The criterion to use in forming flat clusters.
+            See scipy.cluster.hierarchy.fcluster for further details.
+
+        :param depth: int
+            The maximum depth to perform the inconsistency calculation.
+            See scipy.cluster.hierarchy.fcluster for further details.
+
+        :param R: array-like or None
+            The inconsistency matrix to use for the 'inconsistent' criterion.
+            See scipy.cluster.hierarchy.fcluster for further details.
+
+        :param monocrit: array-like or None
+            The statistics upon which non-singleton i is thresholded.
+            See scipy.cluster.hierarchy.fcluster for further details.
+        """
+        self.method = method
+        self.affinity = affinity
+        self.threshold = threshold
+        self.n_clusters = n_clusters
+        self.criterion = criterion
+        self.depth = depth
+        self.R = R
+        self.monocrit = monocrit
+
+    def fit(self, X, y=None):
+        """Perform hierarchical clustering on input data.
+
+        Parameters
+        ----------
+        :param X: array-like, shape (n_samples, n_features) or
+                  (n_samples, n_samples)
+            Input data, as an array of samples or as a distance matrix if
+            affinity='precomputed'.
+
+        Returns
+        -------
+        :returns: self
+        """
+        X = np.array(X)
+
+        if self.affinity == "precomputed":
+            self.linkage_ = hac.linkage(squareform(X), method=self.method)
+
+        else:
+            self.linkage_ = hac.linkage(X,
+                                        method=self.method,
+                                        metric=self.affinity)
+
+        return self
+
+    @property
+    def labels_(self):
+        """Compute the labels assigned to the input data.
+
+        Note that labels are computed on-the-fly from the linkage matrix,
+        based on the value of self.threshold or self.n_clusters.
+        """
```

Comment:
I see your point, but I would rather not complexify this for the time being. If this proves to be a bottleneck, then we will improve it later.